tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1282,8 +1289,16 @@ def _optparse_topic_only(cmd, argv):
     return rosgraph.names.script_resolve_name('rostopic', args[0])
 
 def _rostopic_cmd_type(argv):
-    _rostopic_type(_optparse_topic_only('type', argv))
-    
+    args = argv[2:]
+    from optparse import OptionParser
+    parser = OptionParser(usage="usage: %prog type /topic_or_field", prog=NAME)
+    (options, args) = parser.parse_args(args)
+    if len(args) == 0:
+        parser.error("topic or field must be specified")
+    if len(args) > 1:
+        parser.error("you may only specify one input topic or field")
```

Comment:
This is not a good way of using `OptionParser`. Please use the provided API like `add_option` to declare the expected command line arguments. Then the library will also take care of the error messages automatically.

I would also recommend to use `argparse` since `OptionParser` is deprecated.